File: tmp/scan_missing_pages.py
import os
import re

MODULES_DIR = r"d:\USUARIO\Desktop\complejo\src\modules"

def scan_modules():
    missing_files = []
    
    for module_name in os.listdir(MODULES_DIR):
        module_path = os.path.join(MODULES_DIR, module_name)
        if not os.path.isdir(module_path):
            continue
            
        index_path = os.path.join(module_path, "index.jsx")
        if not os.path.exists(index_path):
            print(f"Module {module_name} is missing index.jsx")
            continue
            
        with open(index_path, "r", encoding="utf-8") as f:
            content = f.read()
            
        # Find imports like: import Something from './pages/Something'
        # Matches './pages/Something', './components/Something', etc.
        imports = re.findall(r"import\s+(\w+)\s+from\s+['\"](\.\/.*?)['\"]", content)
        
        for component_name, import_path in imports:
            # Resolve the path
            # If import_path is './pages/Something', it should be in module_path/pages/Something.jsx (or .js)
            target_path = os.path.join(module_path, import_path.strip("./"))
            
            # Check for .jsx or .js or .tsx or .ts
            found = False
            for ext in [".jsx", ".js", ".tsx", ".ts"]:
                if os.path.exists(target_path + ext):
                    found = True
                    break
                    
            if not found:
                missing_files.append({
                    "module": module_name,
                    "component": component_name,
                    "path": target_path + ".jsx"
                })
                
    return missing_files

if __name__ == "__main__":
    missing = scan_modules()
    if not missing:
        print("No missing files found!")
    else:
        for m in missing:
            print(f"MISSING: {m['path']} (Module: {m['module']}, Component: {m['component']})")

File: tmp/scan_missing_pages.js
const fs = require('fs');
const path = require('path');

const rootDir = process.cwd();
const modulesDir = path.join(rootDir, 'src', 'modules');

function scanModules() {
    const missingFiles = [];
    if (!fs.existsSync(modulesDir)) {
        console.error("Modules directory not found at: " + modulesDir);
        return [];
    }

    const modules = fs.readdirSync(modulesDir);

    for (const moduleName of modules) {
        const modulePath = path.join(modulesDir, moduleName);
        if (!fs.statSync(modulePath).isDirectory()) continue;

        const indexPath = path.join(modulePath, 'index.jsx');
        if (!fs.existsSync(indexPath)) {
            console.log(`Module ${moduleName} is missing index.jsx`);
            continue;
        }

        try {
            const content = fs.readFileSync(indexPath, 'utf-8');
            const importRegex = /import\s+(\w+)\s+from\s+['"](\.\/.*?)['"]/g;
            let match;

            while ((match = importRegex.exec(content)) !== null) {
                const componentName = match[1];
                const importPath = match[2];
                const targetPath = path.join(modulePath, importPath.replace(/^\.\//, ''));
                
                let found = false;
                for (const ext of ['.jsx', '.js', '.tsx', '.ts']) {
                    if (fs.existsSync(targetPath + ext)) {
                        found = true;
                        break;
                    }
                }

                if (!found) {
                    missingFiles.push({
                        module: moduleName,
                        component: componentName,
                        path: targetPath + '.jsx'
                    });
                }
            }
        } catch (err) {
            console.error(`Error reading ${indexPath}: ${err.message}`);
        }
    }
    return missingFiles;
}

try {
    const missing = scanModules();
    if (missing.length === 0) {
        console.log("SUCCESS: No missing files found!");
    } else {
        console.log("FOUND_MISSING_FILES:");
        missing.forEach(m => {
            console.log(`${m.path} | ${m.module} | ${m.component}`);
        });
    }
} catch (e) {
    console.error("CRITICAL_ERROR: " + e.message);
}
